src/Workspaces/Core/Portable/Remote/RemoteFeatureOptions.cs

```diff
@@ -25,6 +23,16 @@ public static bool ShouldComputeIndex(Workspace workspace)
                 case WorkspaceKind.RemoteWorkspace:
                     // Always compute indices in the remote workspace.
                     return true;
+
+                case WorkspaceKind.Host:
+                    // Compute indices in the host workspace when OOP is disabled.
+                    var remoteHostClientService = workspace.Services.GetService<IRemoteHostClientService>();
+                    if (remoteHostClientService is null)
+                    {
+                        return true;
+                    }
+
+                    return !remoteHostClientService.IsEnabled();
```

Comment:
are we sure this is ok?  computing indices in the host is expensive (both CPU and memory).  From perf testing it def affects things like typing latency, and can lead to many more GCs, which impacts the rest of the stuff running in VS.

Reply:
Gotcha.  Though i meant "VS" in the abstract sense.  Namely whatever host we might be in.  But i suppose thems the breaks.  If the host doesn't have an OOP option... they have to live with the perf implication here.

Reply:
@CyrusNajmabadi Some users disabled Roslyn OOP to save system memory even on Windows. This change was a regression for those users which is now being corrected.